src/Components/Components/test/Routing/RouterTest.cs

```diff
@@ -42,72 +42,39 @@ public async Task CanRunOnNavigateAsync()
         {
             // Arrange
             var called = false;
-            async Task OnNavigateAsync(NavigationContext args)
+            Action<NavigationContext> OnNavigateAsync = async (NavigationContext args) =>
             {
                 await Task.CompletedTask;
                 called = true;
-            }
-            _router.OnNavigateAsync = OnNavigateAsync;
-
-            // Act
-            await _renderer.Dispatcher.InvokeAsync(() => _router.RunOnNavigateWithRefreshAsync("http://example.com/jan", false));
-
-            // Assert
-            Assert.True(called);
-        }
-
-        [Fact]
-        public async Task CanHandleSingleFailedOnNavigateAsync()
-        {
-            // Arrange
-            var called = false;
-            async Task OnNavigateAsync(NavigationContext args)
-            {
-                called = true;
-                await Task.CompletedTask;
-                throw new Exception("This is an uncaught exception.");
-            }
-            _router.OnNavigateAsync = OnNavigateAsync;
+            };
+            _router.OnNavigateAsync = new EventCallback<NavigationContext>(null, OnNavigateAsync);
 
             // Act
-            await _renderer.Dispatcher.InvokeAsync(() => _router.RunOnNavigateWithRefreshAsync("http://example.com/jan", false));
+            await _renderer.Dispatcher.InvokeAsync(() => _router.RunOnNavigateAsync("http://example.com/jan", false));
 
             // Assert
             Assert.True(called);
-            Assert.Single(_renderer.HandledExceptions);
-            var unhandledException = _renderer.HandledExceptions[0];
-            Assert.Equal("This is an uncaught exception.", unhandledException.Message);
         }
 
         [Fact]
         public async Task CanceledFailedOnNavigateAsyncDoesNothing()
         {
             // Arrange
             var onNavigateInvoked = 0;
-            async Task OnNavigateAsync(NavigationContext args)
+            Action<NavigationContext> OnNavigateAsync = async (NavigationContext args) =>
             {
                 onNavigateInvoked += 1;
                 if (args.Path.EndsWith("jan"))
                 {
                     await Task.Delay(Timeout.Infinite, args.CancellationToken);
                     throw new Exception("This is an uncaught exception.");
                 }
-            }
-            var refreshCalled = false;
-            _renderer.OnUpdateDisplay = (renderBatch) =>
-            {
-                if (!refreshCalled)
-                {
-                    refreshCalled = true;
-                    return;
-                }
-                Assert.True(false, "OnUpdateDisplay called more than once.");
             };
-            _router.OnNavigateAsync = OnNavigateAsync;
```

Comment:
It will be good to capture the new behavior of OnUpdateDisplay, I imagine that this has changed since we are removing it.

I would suggest we capture on an integer, the number of times it is being called and check against how many times we expect it to be called. That will help catch regressions if in the future we make a change to the implementation that results in a subtle change in the rendering sequence.

Reply:
Added this.